src/GraphQL.Tests/Execution/MutationTests.cs

```diff
@@ -240,7 +240,7 @@ public void evaluates_mutations_correctly_in_the_presense_of_a_failed_mutation()
                 }";
 
             var result = AssertQueryWithErrors(query, expected, root: new Root(6), expectedErrorCount: 2);
-            result.Errors.First().InnerException.Message.ShouldBe("Cannot change the number");
+            result.Errors.First().InnerException.InnerException.Message.ShouldBe("Cannot change the number");
```

Comment:
Looking at how this changes the exception, perhaps we should instead just not try to await the task if it is already faulted instead of re-throw?

```
if (!task.IsFaulted)
{
   await task.ConfigureAwait(false);
   result = task.GetProperyValue("Result");
}
```